spec/haproxy/templates/certs.ttar_spec.rb

```diff
@@ -392,27 +392,66 @@
   end
 
   describe 'ha_proxy.ext_crt_list' do
-    let(:ttar) do
-      template.render({
-        'ha_proxy' => {
-          'crt_list' => [{
-            'ssl_pem' => 'ssl_pem contents'
-          }],
-          'ext_crt_list' => true
-        }
-      })
+    context 'when there are no internal certificates' do
+      let(:ttar) do
+        template.render({
+          'ha_proxy' => {
+            # FIXME: using ext_crt_list: true currently requires crt_list is [], not nil
+            # if there are no 'internal' certs but probably shouldn't require this
+            'crt_list' => [],
```

Comment:
It seems like an unnecessary restriction though, no? -- with empty array it actually works fine, as shown by the acceptance test https://github.com/cloudfoundry-incubator/haproxy-boshrelease/blob/2d558dadfa4cd52eda011b3c9f173126db6f645c/acceptance-tests/https_ext_crt_list_test.go#L246

That test has no internal certs but the request is correctly handled with the external cert

Reply:
I was alluding to the 'FIXME' comment. I think it is nothing that needs fixing.
We *could* put something like
```
if_p("ha_proxy.crt_list") || if_p("ha_proxy.ext_crt_list") do
```
in https://github.com/cloudfoundry-incubator/haproxy-boshrelease/blob/master/jobs/haproxy/templates/haproxy.config.erb#L48

but it would probably also further complicate the logic in https://github.com/cloudfoundry-incubator/haproxy-boshrelease/blob/master/jobs/haproxy/templates/certs.ttar.erb

Reply:
Ah sorry I misunderstood -- I've removed the FIXME.

Yes I agree it would make certs.ttar.erb more complicated